packages/zoltan2/src/algorithms/color/Zoltan2_AlgSerialGreedy.hpp

```diff
@@ -130,8 +130,8 @@ class AlgSerialGreedy : public Algorithm<Adapter>
     // Find max degree, since (max degree)+1 is an upper bound.
     lno_t maxDegree = 0; 
     for (size_t i=0; i<nVtx; i++){
-      if (offsets[i+1]-offsets[i] > maxDegree)
-        maxDegree = offsets[i+1]-offsets[i];
+      if (offsets[i+1]-offsets[i] > static_cast<offset_t>(maxDegree))
+        maxDegree = static_cast<lno_t>(offsets[i+1]-offsets[i]);
```

Comment:
Fixed. I must have been misinterpreting what was actually happening there. Hopefully this is now exactly as you would expect.